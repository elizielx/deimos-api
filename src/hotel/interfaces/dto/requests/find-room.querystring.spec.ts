import { plainToClass } from "class-transformer";
import { validate } from "class-validator";
import { FindRoomsRequestQueryString } from "./find-room.querystring";

describe("FindRoomsRequestQueryString", () => {
    it("should validate a valid request", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: 5,
            take: 10,
        });
        const errors = await validate(request);
        expect(errors.length).toBe(0);
    });

    it("should validate a request with only skip", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: 5,
        });
        const errors = await validate(request);
        expect(errors.length).toBe(0);
    });

    it("should validate a request with only take", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            take: 10,
        });
        const errors = await validate(request);
        expect(errors.length).toBe(0);
    });

    it("should not validate a request with negative skip", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: -1,
            take: 10,
        });
        const errors = await validate(request);
        expect(errors.length).toBeGreaterThan(0);
    });

    it("should not validate a request with zero take", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: 5,
            take: 0,
        });
        const errors = await validate(request);
        expect(errors.length).toBeGreaterThan(0);
    });

    it("should not validate a request with take greater than 20", async () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: 5,
            take: 21,
        });
        const errors = await validate(request);
        expect(errors.length).toBeGreaterThan(0);
    });

    it("matches the snapshot", () => {
        const request = plainToClass(FindRoomsRequestQueryString, {
            skip: 5,
            take: 10,
        });
        expect(request).toMatchSnapshot();
    });
});
