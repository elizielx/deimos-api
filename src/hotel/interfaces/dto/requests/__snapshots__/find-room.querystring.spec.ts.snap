// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindRoomsRequestQueryString matches the snapshot 1`] = `
FindRoomsRequestQueryString {
  "skip": 5,
  "take": 10,
}
`;
